src/stream/rtsp/rtsp_server.rs

```diff
@@ -0,0 +1,186 @@
+        factory.set_buffer_size(0);
+        factory.set_latency(0u32);
+        // factory.set_stop_on_disconnect(false);
+        factory.set_transport_mode(RTSPTransportMode::PLAY);
+        // factory.set_suspend_mode(RTSPSuspendMode::Pause);
+        factory.set_protocols(RTSPLowerTrans::UDP | RTSPLowerTrans::UDP_MCAST);
+
+        let socket_path = format!("/tmp/{path}");
+        let description = format!(
+            concat!(
+                "shmsrc socket-path={socket_path} do-timestamp=true",
+                " ! queue leaky=downstream flush-on-eos=true max-size-buffers=0",
+                " ! application/x-rtp,media=video,clock-rate=90000,payload=96,encoding-name=H264",
+                " ! rtph264depay",
+                " ! rtph264pay name=pay0 aggregate-mode=zero-latency config-interval=10 pt=96",
+            ),
+            socket_path = socket_path,
+        );
+        factory.set_launch(&description);
+
+        if let Some(server) = rtsp_server
+            .path_to_factory
+            .insert(path.to_string(), factory)
+        {
+            return Err(anyhow!("Error: required path already exists! The older was updated with the new configurations: {server:#?}"));
+        }
+
+        Ok(())
+    }
+
+    pub fn start_pipeline(path: &str) -> Result<()> {
+        RTSPServer::configure("0.0.0.0", 8554);
+
+        let mut rtsp_server = RTSP_SERVER.as_ref().lock().unwrap();
+
+        // Much like HTTP servers, RTSP servers have multiple endpoints that
+        // provide different streams. Here, we ask our server to give
+        // us a reference to his list of endpoints, so we can add our
+        // test endpoint, providing the pipeline from the cli.
+        let mounts = rtsp_server
+            .server
+            .mount_points()
+            .context("Could not get mount points")?;
+
+        let factory = rtsp_server.path_to_factory.get(path).unwrap();
+
+        // Now we add a new mount-point and tell the RTSP server to serve the content
+        // provided by the factory we configured above, when a client connects to
+        // this specific path.
+        mounts.add_factory(path, factory);
+
+        rtsp_server.run = true; // start the main loop thread
+
+        Ok(())
+    }
+
+    pub fn stop_pipeline(path: &str) -> Result<()> {
+        let mut rtsp_server = RTSP_SERVER.as_ref().lock().unwrap();
+
+        // Much like HTTP servers, RTSP servers have multiple endpoints that
+        // provide different streams. Here, we ask our server to give
+        // us a reference to his list of endpoints, so we can add our
+        // test endpoint, providing the pipeline from the cli.
+        let mounts = rtsp_server
+            .server
+            .mount_points()
+            .context("Could not get mount points")?;
+
+        mounts.remove_factory(path);
+
+        rtsp_server.path_to_factory.remove(path);
+        // TODO: call mainloop.quit() to stop the server if there is no endpoints
```

Comment:
is that really necessary ?
can't we have it running ? It should use no cpu at all and the code will be simpler.

Reply:
Indeed its not really necessary :)